src/client/components/WithModal.tsx

```diff
@@ -29,10 +31,14 @@ const withModal = <T extends ElementType, U>(Tag: T, ModalTag: ComponentType<U>)
           return altClick();
         }
 
+        if (stopProgagation) {
+          event.stopPropagation();
+        }
+
         event.preventDefault();
         return toggle();
       },
-      [altClick, toggle],
+      [altClick, stopProgagation, toggle],
     );
 
     return (
```

Comment:
Down here where <Tag ..> is setup the onClick is set to handleClick. So even if allProps had onClick, such as because Button supports onClick, its value won't be used.